models/User_test.go

```diff
@@ -773,6 +774,75 @@ func Test_PerformBuyFromExchangeTransaction(t *testing.T) {
 	}
 }
 
+// Helper for the deadlock test written below. If you want to test any table OTHER than TransactionSummary,
+// all you need to do is update this helper function. The actual Test_Deadlock function need not be updated.
+func saveTransSum(userID int, t *testing.T) {
+	db := getDB()
+	tx := db.Begin()
+
+	transactionSummary := &TransactionSummary{
+		UserId:        uint32(userID),
+		StockId:       2,
+		StockQuantity: 10,
+		Price:         12.5,
+	}
+
+	if err := tx.Save(transactionSummary).Error; err != nil {
+		t.Errorf("Error updating the transaction summary. Rolling back. Error : +%v", err)
+		tx.Rollback()
+		return
+	}
+
+	if err := tx.Commit().Error; err != nil {
+		t.Errorf("Error committing the transaction. Failing. %+v", err)
+		tx.Rollback()
+		return
+	}
+
+	t.Logf("TransactionSummary table updated successfully - %+v", transactionSummary)
+}
+
+// This is a test for deadlock when multiple goroutines spam insert statements into the TransactionSummary table.
+// However, with a few modifications, it can be made to test any table for deadlock possibilities. Feel free to modify
+// this test if there's any other table you'd like to test insert statements on.
+func Test_Deadlock(t *testing.T) {
+
+	db := getDB()
+	numUsers := 10
+
+	for i := 1; i <= numUsers; i++ {
+		user := &User{Id: uint32(i), Email: fmt.Sprintf("user%v@deadlock.com", i), Cash: 1000000}
```

Comment:
fmt isn't used anywhere instead logger is used

Reply:
I'm not logging here, I'm formatting a string (i.e) I'm not using fmt to print anything.